packages/core/src/components/overflow-list/overflowList.tsx

```diff
@@ -118,14 +141,28 @@ export class OverflowList<T> extends React.PureComponent<IOverflowListProps<T>,
         ) {
             // reset visible state if the above props change.
             this.setState({
+                direction: OverflowDirection.GROW,
+                lastOverflow: undefined,
                 overflow: [],
                 visible: nextProps.items,
             });
         }
     }
 
-    public componentDidUpdate() {
+    public componentDidUpdate(_prevProps: IOverflowListProps<T>, prevState: IOverflowListState<T>) {
         this.repartition(false);
+        if (
+            // if a resize operation has just completed (transition to NONE)
+            this.state.direction === OverflowDirection.NONE &&
+            this.state.direction !== prevState.direction &&
+            // shrink operation does not require checking last items
+            (prevState.direction === OverflowDirection.SHRINK ||
+                // if current overflow is different from last completed state
+                this.state.lastOverflow == null ||
```

Comment:
Could we do `=== undefined`, or is `== null` just idiomatic for the codebase?

Reply:
idiomatic